countdown/database.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+from datetime import datetime
+
+import discord
+from sqlalchemy import BigInteger, Column, Integer, String, DateTime
+from typing import List, Dict, Optional
+
+from pie.database import database, session
+
+
+class CountdownItem(database.base):
+    """Represents a database Countdown item for :class:`Reminder` module.
+
+    Attributes:
+        idx: The database ID.
+        guild_id: ID of the guild.
+        author_id: User ID of reminder author.
+        permalink: Message URL.
+        message: Reminder text (None if empty).
+        origin_date: Date of creation.
+        countdown_date: Date for reminding
+        status: Status of reminder
+    """
+
+    __tablename__ = "fun_countdown_countdownitem"
+
+    idx = Column(Integer, primary_key=True, autoincrement=True)
+    guild_id = Column(BigInteger, default=None)
+    author_id = Column(BigInteger, default=None)
+    name = Column(String, default=None)
+    permalink = Column(String, default=None)
+    message = Column(String, default=None)
+    origin_date = Column(DateTime(timezone=True), default=None)
+    countdown_date = Column(DateTime(timezone=True), default=None)
+
+    @staticmethod
+    def add(
+        author: discord.Member,
+        name: str,
+        permalink: str,
+        message: str,
+        origin_date: datetime,
+        countdown_date: datetime,
+    ) -> CountdownItem:
+        """Creates a new CountdownItem in the database.
+
+        Args:
+            author: Countdown author.
+            name: Event name
+            permalink: URL of countdown message.
+            message: Countdown text (None if empty).
+            origin_date: Date of creation.
+            countdown_date: Date for reminding
+
+        Raises:
+            ValueError: End time already passed.
+
+        Returns:
+            The created database item.
+        """
+
+        origin_date = datetime.now()
+        if countdown_date < origin_date:
+            raise ValueError
+
+        item = CountdownItem(
+            guild_id=author.guild.id if hasattr(author, "guild") else 0,
+            author_id=author.id,
+            name=name,
+            permalink=permalink,
+            message=message,
+            origin_date=origin_date,
+            countdown_date=countdown_date,
+        )
+
+        session.add(item)
+        session.commit()
+        return item
+
+    @staticmethod
+    def get(author_id: int, name: str) -> Optional[CountdownItem]:
+        query = (
+            session.query(CountdownItem)
+            .filter_by(author_id=author_id, name=name)
+            .one_or_none()
+        )
+        return query
+
+    @staticmethod
+    def get_all(
+        guild: discord.Guild = None,
+        idx: int = None,
```

Comment:
1/ Do we need to filer by idx?
2/ I'd prefer if database files worked directly with `guild_id`, not DIscord objects. This is different from how the rest of the codebase is done.